spec/cancan/model_adapters/active_record_adapter_spec.rb

```diff
@@ -237,101 +237,84 @@ class User < ActiveRecord::Base
     end
 
     it 'has false conditions if no abilities match' do
-      expect(@ability.model_adapter(Article, :read).conditions).to eq("'t'='f'")
+      expect(@ability.model_adapter(Article, :read).database_records.to_sql.strip.squeeze(' '))
+        .to eq(%q(SELECT "articles".* FROM "articles" WHERE (1 == 0)))
     end
 
     it 'returns false conditions for cannot clause' do
       @ability.cannot :read, Article
-      expect(@ability.model_adapter(Article, :read).conditions).to eq("'t'='f'")
+      expect(@ability.model_adapter(Article, :read).database_records.to_sql.strip.squeeze(' '))
+        .to eq(%q(SELECT "articles".* FROM "articles" WHERE (1 == 0)))
     end
 
     it 'returns SQL for single `can` definition in front of default `cannot` condition' do
       @ability.cannot :read, Article
       @ability.can :read, Article, published: false, secret: true
-      expect(@ability.model_adapter(Article, :read).conditions)
-        .to orderlessly_match(%("#{@article_table}"."published" = 'f' AND "#{@article_table}"."secret" = 't'))
+      expect(@ability.model_adapter(Article, :read).database_records.to_sql.strip.squeeze(' ').gsub(' )', ')'))
```

Comment:
IIRC this was because AR3 and AR4 adapters produced queries that differed only in whitespace, so I normalized them to a standard form. With the consolidation of the AR drivers I'll revisit this and remove if possible